documentcloud/organizations/models.py

```diff
@@ -138,4 +140,47 @@ def use_ai_credits(self, amount):
         organization.monthly_ai_credits -= ai_credit_count["monthly"]
         organization.number_ai_credits -= ai_credit_count["regular"]
         organization.save()
+
+        organization.ai_credit_logs.create(
+            user_id=user_id,
+            organization=organization,
```

Comment:
I see organization gets referenced on line 129- can you explain to me how that line works?

Reply:
https://docs.djangoproject.com/en/4.2/ref/models/querysets/#select-for-update

It reloads the self object with a lock, so that it doesn't try to run two of these calls in parallel, which are not thread safe.  Can explain more on a call if needed.

Reply:
Makes sense. I don't see anything that stands out to me otherwise.